common/trieForDirPath.go

```diff
@@ -0,0 +1,72 @@
+package common
+
+import (
+	"strings"
+)
+
+type TrieNode struct {
+	dirPathSeg             map[string]*TrieNode
+	status                 uint32
```

Comment:
Upon further thinking, the same problem returns to existence prior to the job plan tracker. When AzCopy gets restarted, e.g. `jobs resume`, we don't persist these brand new folders we created into the plan file.

Reply:
Separate comment, because unrelated to this PR, but future work, wrt scaling, part of the equation is going to be reducing plan file size after we finish sets of transfers. We'll want some efficient way to keep track of the folder structure we've created down the line. This is more of a TODO for me than you.

Reply:
Back to the comment re: unregisteredButCreated. If we're adding these folders into the plan file, we can just set status to the index in the plan file. Since we're never dropping things from the trie to save memory now, we should also worry about adding folders we do register in the plan files into this trie, even if the tx is already completed.